python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -960,14 +971,15 @@ def _with_header(msg: str) -> str:
 
             raise SerdesUsageError(_with_header(error_msg))
 
-        value_param = value_params[index]
-        if value_param.name != field:
-            error_msg = (
-                "Params to __new__ must match the order of field declaration in the namedtuple. "
-                f'Declared field number {index + 1} in the namedtuple is "{field}". '
-                f'Parameter {index + 1} in __new__ method is "{value_param.name}".'
-            )
-            raise SerdesUsageError(_with_header(error_msg))
+        if require_args_to_match_field_ordering:
+            value_param = value_params[index]
+            if value_param.name != field:
+                error_msg = (
```

Comment:
@alangenfeld do you know if we can remove this check altogether? Looks like when rebuild named tuples during deserialization we use kwargs, so this might not be needed anymore

Reply:
heres the history i could dig up, unfortunate we are locked out of our phabricator content now
https://github.com/dagster-io/dagster/commit/52be7e8a94c4379c16132ab511cbfb007bdbce1e
https://github.com/dagster-io/dagster/issues/2372

I **think** `PipelineRun` was getting pickled when passed around in `multiprocessing` contexts.

Reply:
* might be better to rename the toggling bool to something along the lines of `cant_pickle` to more clearly advertise what opting out of the protection represents 
* probably want to put pickle round tripping under test to demonstrate that it is known to fail in whatever way it does (if possible make it fail clearly)

Reply:
thanks for the context here.

I renamed the param to `is_pickleable`. I also overrode a `__getstate__` method on `TimeWindowPartitionsDefinition` that enables raising an error when attempting to pickle:

https://docs.python.org/3/library/pickle.html#handling-stateful-objects